### PR TITLE
TR-122: rolledup/work-status

### PR DESCRIPTION
**Pull request by binaryberserker**

### CODE REVIEW

File: public/jira/rolledup-and-rolledback/rollup-and-rollback.ts [hunk 1]
```diff
@@ -1,33 +1,50 @@
...
 import { JiraIssue, RollupLevelAndCalculation } from "../shared/types";
 
-type ExtractIssueOrReleaseType<T> = T extends IssueOrRelease<infer X> ? X : never;
-
-export type WithIssueLastPeriod<T> = {
+export type WithIssueLastPeriod<T = unknown> = {
   issueLastPeriod: IssueOrRelease<T>;
 };
 
+export type WithRollups = WithReportingHierarchy &
```
Comment (bmomberger-bitovi) [inline public/jira/rolledup-and-rolledback/rollup-and-rollback.ts:29]: not a blocker but is this exported type actually used anywhere else? i can't find it imported anywhere in the PR

**Review by bmomberger-bitovi — APPROVED** [review]

File: public/jira/rolledup-and-rolledback/rollup-and-rollback.ts (hunk 1 above)
Comment (binaryberserker) [reply public/jira/rolledup-and-rolledback/rollup-and-rollback.ts:29]: it's used on line 47 of this file, I think I was using it somewhere else in a previous version, and I left it exported cuz it seemed like something we were likely to need elsewhere

File: public/timeline-report.js [hunk 2]
```diff
@@ -14,8 +14,8 @@ import "./select-report-type/select-report-type.js";
 import "./select-view-settings/select-view-settings.js";
 
 import { rollupAndRollback } from "./jira/rolledup-and-rolledback/rollup-and-rollback";
-import { calculateReportStatuses } from "./jira/rolledup/work-status.js/work-status.js";
-import { groupIssuesByHierarchyLevelOrType } from "./jira/rollup/rollup.js";
+import { calculateReportStatuses } from "./jira/rolledup/work-status.js/work-status";
+import { groupIssuesByHierarchyLevelOrType } from "./jira/rollup/rollup";
 import { pushStateObservable } from "./shared/state-storage.js";
```
Comment (binaryberserker) [inline public/timeline-report.js:19]: had a merge conflict here, i removed the .js extensions and main added pushStateObservable, used main and re-removed the extensions

File: public/timeline-report.js [hunk 3]
```diff
@@ -14,8 +14,8 @@ import "./select-report-type/select-report-type.js";
 import "./select-view-settings/select-view-settings.js";
 
 import { rollupAndRollback } from "./jira/rolledup-and-rolledback/rollup-and-rollback";
-import { calculateReportStatuses } from "./jira/rolledup/work-status.js/work-status.js";
-import { groupIssuesByHierarchyLevelOrType } from "./jira/rollup/rollup.js";
+import { calculateReportStatuses } from "./jira/rolledup/work-status/work-status";
+import { groupIssuesByHierarchyLevelOrType } from "./jira/rollup/rollup";
 import { pushStateObservable } from "./shared/state-storage.js";
```
Comment (binaryberserker) [inline public/timeline-report.js:19]: had a merge conflict here, i removed the .js extensions and main added pushStateObservable, used main and re-removed the extensions

Comment (bmomberger-bitovi) [reply public/timeline-report.js:19]: 🆗

File: public/jira/rolledup-and-rolledback/rollup-and-rollback.ts (hunk 1 above)
Comment (bmomberger-bitovi) [reply public/jira/rolledup-and-rolledback/rollup-and-rollback.ts:29]: It seems like the kind of type that might be better served when pulled out to a common types file, _if_ you're using it externally.

**Pull request merged**